pkg/fqdn/cache.go

```diff
@@ -154,16 +190,124 @@ func (c *DNSCache) updateWithEntry(entry *cacheEntry) {
 		c.forward[entry.Name] = entries
 	}
 	c.updateWithEntryIPs(entries, entry)
+	if c.perHostLimit > 0 && len(entries) > c.perHostLimit {
```

Comment:
can you move this to after the removeExpired below?

Reply:
done!